VAMobile/src/screens/BenefitsScreen/BenefitsScreen.tsx

```diff
@@ -37,14 +38,32 @@ const BenefitsScreen: FC<BenefitsScreenProps> = () => {
   const ratingIsDefined = ratingPercent !== undefined && ratingPercent !== null
   const combinedPercentText = ratingIsDefined ? t('disabilityRating.combinePercent', { combinedPercent: ratingPercent }) : undefined
 
-  const claimsDestination = featureEnabled('decisionLettersWaygate') && userAuthorizedServices?.decisionLetters ? 'Claims' : 'ClaimsHistory'
+  const onDisabilityRatings = () => {
+    if (waygateNativeAlert('WG_DisabilityRatings')) {
```

Comment:
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7278

Created a code upkeep ticket for this. Need to ask some bigger picture questions regarding event capture analytics and setting up a best practice for individual screen navs.

Reply:
I like that idea a lot. It would prevent needing to have this logic is so many places, and we wouldn't need to remember to add this check moving forward. We have places where this check is wrapped around `navigation.navigate` calls, so we'd need to migrate those to use the same function if we didn't want to repeat the logic there also. Although, I think places that use `navigation.navigate` have to use that for analytics.

Instead of needing a new function, I wonder if we could add this logic directly to the [function](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/3faf9839110103962a669475a95b1962014595f2/VAMobile/src/utils/hooks/index.tsx#L101-L103) that's returned from `useRouteNavigation` hook, that way we wouldn't even need to update any `navigateTo` function calls